### DISCUSSION TITLE
No Control Input but the Altitude comes to minus infinity!!! (JSBSim with Matlab)

### DISCUSSION THREAD

**Discussion by BadboyIntheLoop** [JSBSim-Team/jsbsim#610] [Support requests]
**Dear JSBSim Pro!!!**
I've met a problem when using JSBSim_Sfunction in matlab simulink!  When I take no control inputs (Throttle, Aileron, Elevator, Rudder = 0), the altitude comes to minus infinity ($$ h \to -\infty $$).  I tested with B747 and c172p and met this error. I think there might be no ground floor (ground floor at minus infinity). I saw Prof.Agostino can run JSBSim in Matlab simulink smoothly. Can I set altitude of ground floor at 0.0, so that altitude always equals to zero for above situation!

Please help me for this problem. Many thanks!!!

**Comment by bcoconni** [comment]
This topic has been moved to JSBSim Discussions as per the issue template statement: `Issues that are neither bug reports or feature requests will be transferred to` [Discussions](https://github.com/JSBSim-Team/jsbsim/discussions) `by the team and the issue will be locked and closed.`.

Support for this topic will therefore continue within this discussion. Thanks for your understanding.

**Reply by bcoconni** [reply]
@BadboyIntheLoop, unless you can provide reproducible elements that prove that JSBSim is not behaving the way it should, problems reported by users are considered support requests and as such should be discussed in [Discusions](https://github.com/JSBSim-Team/jsbsim/discussions).

**Comment by bcoconni** [comment]
> I think there might be no ground floor (ground floor at minus infinity). I saw Prof.Agostino can run JSBSim in Matlab simulink smoothly. Can I set altitude of ground floor at 0.0, so that altitude always equals to zero for above situation!

By default, JSBSim automatically sets the ground at altitude zero. I'd guess that either you are using models that have no contacts and that the aircraft gears are up which prevents JSBSim from detecting contact with the ground.

**Reply by TatChung68** [reply]
Sorry for my post at the Issue. With my problem I purely use default model in ./aircraft/B747 without changes, I also think altitude default init at 0 but it fails. I have questions about must do trim firstly like using script instead of Matlab? Thank you

**Reply by bcoconni** [reply]
We need more details to help you. How is your model initialized ? Have you checked that the gears are down ?

**Reply by BadboyIntheLoop** [reply]
![image](https://user-images.githubusercontent.com/100922984/162857146-341c34c6-7e0d-4bc9-aa92-2dd86cc8eb0e.png)
![image](https://user-images.githubusercontent.com/100922984/162857061-26cfc00a-6224-48d2-886c-15ec6f8469e0.png)
![image](https://user-images.githubusercontent.com/100922984/162857110-0cc7bfc1-8051-4333-86a8-db9c2ed0ac67.png)

Yepp, the first picture is my simulink, I set all control inputs to 0. After that, I sim that simulink as you can see, it run normally in second image and result in third image, altitude comes to minus infinity!!!

**Comment by seanmcleod** [comment]
> I also think altitude default init at 0

And looking at the SimuLink graph, yes it looks like the altitude starts off at 0 and then it just sinks into the ground.

As @bcoconni asked, have you checked the model for `contacts` and in particular have you initialized the aircraft with the gear down or up?

```xml
    <ground_reactions>
        <contact type="BOGEY" name="NOSE_LG">
            <location unit="IN">
                <x> 396 </x>
                <y> 0 </y>
                <z> -206 </z>
            </location>
            <static_friction> 0.8 </static_friction>
            <dynamic_friction> 0.5 </dynamic_friction>
            <rolling_friction> 0.02 </rolling_friction>
            <spring_coeff unit="LBS/FT"> 22000 </spring_coeff>
            <damping_coeff unit="LBS/FT/SEC"> 87302.6 </damping_coeff>
            <max_steer unit="DEG"> 5 </max_steer>
            <brake_group> NONE </brake_group>
            <retractable>1</retractable>
        </contact>

```

The 747 model you're using only has ground contacts for the landing gear, so if they are retracted (not deployed) then there will be no ground reaction and the aircraft will fall through the earth, which is what it looks like is happening to you.

**Reply by BadboyIntheLoop** [reply]
Sorry for my late reply!!! Basically, I fixed this problem by instuctions from you and mr.@bcoconni!!! But I meet a new problem with model c172p aircraft, when no control input (with gear =1 always) the 12-states (u,v,w, h ....) come to nan in matlab!!! It's not quite neccesary with me, I only mention it here!

**Reply by bcoconni** [reply]
When gears are down, you should spawn the aircraft with an altitude high enough to allow room for the landing gears height. Otherwise, the aircraft is initialized with landing gears compressed to an insane value and the aircraft is catapulted into space when the simulation is launched.

For the `c172p` you should start with an altitude of approximately 4 ft or a bit higher: the aircraft will then fall and gently reach contact with the ground. An initial altitude of 0 ft means that the gears lengths are compressed by 4 ft and that the gears springs have accumulated a huge compression energy which, when released, apply an insane force to the aircraft, hence the NaNs.

**Reply by BadboyIntheLoop** [reply]
Yepp Thank You!!! I saw it but if applying a throttle-norm equals to 1 (max throttle), I don't see the aircraft (B747 or c127p) move that means u = 0, h = const, can you explain this situation for me!!!

**Reply by bcoconni** [reply]
This is most likely due to the engines not running.

**Reply by BadboyIntheLoop** [reply]
Oh really! I'll try it, I don't think engines need to enable
